ui-v2/src/components/events/events-timeline/events-timeline.tsx

```diff
@@ -74,28 +75,40 @@ function TimelinePoint({ event, isLast }: { event: Event; isLast: boolean }) {
 	);
 }
 
+/**
+ * Formats a Date object as YYYY-MM-DD for use in route parameters.
+ */
+function formatRouteDate(date: Date): string {
+	const year = date.getFullYear();
+	const month = String(date.getMonth() + 1).padStart(2, "0");
+	const day = String(date.getDate()).padStart(2, "0");
+	return `${year}-${month}-${day}`;
+}
+
 function EventNameWithPrefixes({
+	eventId,
 	eventName,
+	occurred,
 	onEventClick,
 }: {
+	eventId: string;
 	eventName: string;
+	occurred: string;
 	onEventClick?: (eventName: string) => void;
 }) {
 	const label = formatEventLabel(eventName);
+	const eventDate = formatRouteDate(new Date(occurred));
 
 	return (
 		<div className="flex flex-col gap-0.5">
-			{onEventClick ? (
-				<button
-					type="button"
-					onClick={() => onEventClick(eventName)}
-					className="text-left font-medium hover:underline"
-				>
-					{label}
-				</button>
-			) : (
-				<span className="font-medium">{label}</span>
-			)}
+			<Link
+				to="/events/event/$eventDate/$eventId"
+				params={{ eventDate, eventId }}
+				className="text-left font-medium hover:underline"
+				onClick={onEventClick ? () => onEventClick(eventName) : undefined}
```

Comment:
We don't need `onEventClick` on these links. Remove it and remove the `onEventClick` prop altogether.